File: LeetCode/00009. Palindrome Number/00009.c
bool isPalindrome(int x){
    unsigned int tmp = 0, y = x;
    
    if (x < 0)
        return false;

    if (y) {
        tmp = y % 10;
        y /= 10;
    }
    
    while (y) {
        tmp *= 10;
        tmp += y % 10;
        y /= 10;
    }
    
    if (tmp == x)
        return true;
    return false;
}

File: LeetCode/00009. Palindrome Number/00009.java
// https://leetcode.com/problems/palindrome-number

class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0) {
            return false;
        }

        int rev = 0, tmp = x;
        
        while (tmp > 0) {
            rev = rev * 10 + tmp % 10;
            tmp /= 10;
        }

        return rev == x;
    }
}
